community/kernel/src/main/java/org/neo4j/kernel/impl/util/HexPrinter.java

```diff
@@ -307,7 +307,7 @@ public static String hex( ByteBuffer bytes, int offset, int length )
      */
     public static String hex( ByteBuffer bytes )
     {
-        return hex( bytes, 0, bytes.position() );
+        return hex( bytes, 0, bytes.limit() );
```

Comment:
Hmm, I didn't notice that diff, good point. 

IMO, this is the right way to do it - operations on bytebuffers act from it's current position and to its limit, eg. the behavior of append is inconsistent with standard practice. Flip is a natural part of working with BBs.

I'd rather shift the behavior of append, if that's ok.